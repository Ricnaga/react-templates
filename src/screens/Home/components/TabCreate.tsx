import { useAxios } from '@application/api/axios/useAxios';
import {
  Button,
  FormControl,
  FormLabel,
  Grid,
  Input,
  useToast,
} from '@chakra-ui/react';
import { useFormik } from 'formik';

enum FormikValues {
  NAME = 'name',
}

type FormikSubmit = Record<'name', string>;

export function TabCreate() {
  const toast = useToast();
  const {
    functions: { callPOST },
  } = useAxios();

  const formik = useFormik({
    initialValues: {
      [FormikValues.NAME]: '',
    },
    onSubmit: async (values: FormikSubmit) =>
      callPOST({
        url: '/user',
        bodyData: {
          id: Math.random().toString(),
          name: values.name,
        },
      })
        .then(() =>
          toast({
            title: 'Success',
            description: 'This user was created',
            status: 'success',
            isClosable: true,
          }),
        )
        .catch(() =>
          toast({
            title: 'Error',
            description: 'This user was not created',
            status: 'error',
            isClosable: true,
          }),
        ),
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <Grid gap={6} maxW="800px" m="auto" mt={10}>
        <FormControl>
          <FormLabel>Name</FormLabel>
          <Input
            type="text"
            placeholder="Insert a name."
            size="lg"
            name={FormikValues.NAME}
            onChange={formik.handleChange}
          />
        </FormControl>

        <Button colorScheme="telegram" type="submit">
          Criar
        </Button>
      </Grid>
    </form>
  );
}
