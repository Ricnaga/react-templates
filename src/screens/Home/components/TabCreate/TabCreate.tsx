import { Button, FormControl, FormLabel, Grid, Input } from '@chakra-ui/react';
import { useFormik } from 'formik';

import { useAxios } from '@application/api/axios/useAxios';
import { useSnackbar } from '@shared/components/Snackbar/useSnackbar';

enum FormikValues {
  NAME = 'name',
}

type FormikSubmit = Record<'name', string>;

type TabCreateProps = {
  handleUpdateUsers: () => void;
};

export function TabCreate({ handleUpdateUsers }: TabCreateProps) {
  const { onSnackBar } = useSnackbar();
  const {
    functions: { callPOST },
  } = useAxios();

  const formik = useFormik({
    initialValues: {
      [FormikValues.NAME]: '',
    },
    onSubmit: async (values: FormikSubmit) =>
      callPOST({
        url: '/user',
        bodyData: {
          id: Math.random().toString(),
          name: values.name,
        },
      })
        .then(() => handleUpdateUsers())
        .then(() => onSnackBar('This user was created'))
        .catch(() => onSnackBar('This user was not created', 'error')),
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <Grid gap={6} maxW="800px" m="auto" mt={10}>
        <FormControl>
          <FormLabel>Name</FormLabel>
          <Input
            type="text"
            placeholder="Insert a name."
            size="lg"
            name={FormikValues.NAME}
            onChange={formik.handleChange}
          />
        </FormControl>

        <Button colorScheme="telegram" type="submit">
          Criar
        </Button>
      </Grid>
    </form>
  );
}
