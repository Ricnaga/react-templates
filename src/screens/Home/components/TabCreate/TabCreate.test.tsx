import { renderWithTheme } from '@application/test/testing-library';

import { TabCreate } from './TabCreate';

let containerElement: HTMLElement;

describe('Component: TabCreate', () => {
  beforeEach(async () => {
    const { container } = renderWithTheme(
      <TabCreate handleUpdateUsers={jest.fn} />,
    );

    containerElement = container;
  });

  it('should render correctly', () => {
    expect(containerElement).toMatchSnapshot();
  });
});
