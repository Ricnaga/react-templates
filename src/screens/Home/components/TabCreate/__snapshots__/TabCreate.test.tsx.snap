// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TabCreate should render correctly 1`] = `
<div>
  <form>
    <div
      class="css-c19cha"
    >
      <div
        class="chakra-form-control css-1kxonj9"
        role="group"
      >
        <label
          class="chakra-form__label css-g6pte"
          for="field-:r0:"
          id="field-:r0:-label"
        >
          Name
        </label>
        <input
          class="chakra-input css-1qrx3zr"
          id="field-:r0:"
          name="name"
          placeholder="Insert a name."
          type="text"
        />
      </div>
      <button
        class="chakra-button css-8bo503"
        type="submit"
      >
        Criar
      </button>
    </div>
  </form>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
